### PR TITLE
DROOLS-1565 Implement a KieScanner listener

### PR DESCRIPTION
**Pull request by tarilabs**

### CODE REVIEW

File: drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieScanner.java [hunk 1]
```diff
@@ -25,10 +25,6 @@
 
 public interface InternalKieScanner extends KieScanner {
 
-    enum Status {
```
Comment (tarilabs) [inline drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieScanner.java:28]: Moved up to `KieScanner`.

File: kie-ci/pom.xml [hunk 2]
```diff
@@ -150,6 +150,10 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>commons-io</groupId>
+      <artifactId>commons-io</artifactId>
```
Comment (tarilabs) [inline kie-ci/pom.xml:155]: This was a transitive dependency, now needed also directly.

**Comment by tarilabs** [comment]
Depends on kiegroup/droolsjbpm-knowledge#232 .

File: kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java [hunk 3]
```diff
@@ -224,6 +244,175 @@ public void testKScannerWithRange() throws Exception {
...
+        assertThat( events, CoreMatchers.hasItem( new KieScannerStatusChangeEventImpl(KieScanner.Status.SCANNING) ) );
+        assertThat( events, CoreMatchers.not( CoreMatchers.hasItem( new KieScannerStatusChangeEventImpl(KieScanner.Status.UPDATING) )) );
+        events.clear();
+        
+        repository.installArtifact(releaseId1, kJar1, createKPom(fileManager, releaseId1));
+
+        KieSession ksession = kieContainer.newKieSession("KSession1");
+        checkKSession(ksession, "rule1", "rule2");
+
+        // Create faulty KJAR for
```
Comment (tkobayas) [inline kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java]: Just to understand the test case... This is not a faulty KJAR, right?

**Review by tkobayas — APPROVED** [review]
Thank you @tarilabs , that looks great. A client can get the result from the event listener.

File: kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java (hunk 3 above)
Comment (tarilabs) [reply kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java]: Well spotted, I must inadvertently placed this comment in the wrong place. Thank you!

**Pull request merged**